packages/dd-trace/src/appsec/sdk/track_event.js

```diff
@@ -63,7 +69,112 @@ function trackCustomEvent (tracer, eventName, metadata) {
 
   trackEvent(eventName, metadata, 'trackCustomEvent', getRootSpan(tracer))
 
-  incrementSdkEventMetric('custom')
+  incrementSdkEventMetric('custom', 'v1')
+
+  if (eventName === 'users.login.success' || eventName === 'users.login.failure') {
+    runWaf(eventName)
+  }
+}
+
+function trackUserLoginSuccessV2 (tracer, login, user, metadata) {
+  if (!login || typeof login !== 'string') {
+    log.warn('[ASM] Invalid login provided to eventTrackingV2.trackUserLoginSuccess')
+    return
+  }
+
+  incrementSdkEventMetric('login_success', 'v2')
+
+  const rootSpan = getRootSpan(tracer)
+  if (!rootSpan) {
+    log.warn('[ASM] Root span not available in eventTrackingV2.trackUserLoginSuccess')
+    return
+  }
+
+  const wafData = { login }
+
+  metadata = {
+    'usr.login': login,
+    ...metadata
+  }
+
+  if (user) {
+    if (typeof user !== 'object') {
+      user = { id: user }
+    }
+
+    if (user.id) {
+      wafData.id = user.id
+      setUserTags(user, rootSpan)
+      metadata.usr = user
+    }
+  }
+
+  trackEvent('users.login.success', metadata, 'eventTrackingV2.trackUserLoginSuccess', rootSpan)
+
+  runWaf('users.login.success', wafData)
+}
+
+function trackUserLoginFailureV2 (tracer, login, exists, metadata) {
+  if (!login || typeof login !== 'string') {
+    log.warn('[ASM] Invalid login provided to eventTrackingV2.trackUserLoginFailure')
+    return
+  }
+
+  incrementSdkEventMetric('login_failure', 'v2')
+
+  const rootSpan = getRootSpan(tracer)
+  if (!rootSpan) {
+    log.warn('[ASM] Root span not available in eventTrackingV2.trackUserLoginFailure')
+    return
+  }
+
+  const wafData = { login }
+
+  if (typeof exists === 'object' && metadata === undefined) {
+    metadata = exists
+    exists = false
+  }
+
+  metadata = {
+    'usr.login': login,
+    'usr.exists': exists ? 'true' : 'false',
+    ...metadata
+  }
+
+  trackEvent('users.login.failure', metadata, 'eventTrackingV2.trackUserLoginFailure', rootSpan)
+
+  runWaf('users.login.failure', wafData)
+}
+
+function flattenFields (fields, sdkMethodName, depth = 0) {
```

Comment:
sdkMethodName is only used to be passed to itself no ?

Reply:
by that i mean it's not used anywhere, and the linter is not detecting it because it's recursive

Reply:
true, it was there because the log was originally into the method. removing